binaries/daemon/src/coordinator.rs

```diff
@@ -81,6 +81,29 @@ pub async fn register(
                 timestamp,
             } = event;
             let (reply_tx, reply_rx) = oneshot::channel();
+
+            // Respond to the coordinator's destroy event immediately as the daemon is about to destroy itself.
+            match event {
+                DaemonCoordinatorEvent::Destroy => {
+                    tracing::info!("Received destroy event from dora-coordinator");
+                    let serialized =
+                        match serde_json::to_vec(&DaemonCoordinatorReply::DestroyResult(Ok(())))
+                            .wrap_err("failed to serialize DaemonCoordinatorReply")
+                        {
+                            Ok(r) => r,
+                            Err(err) => {
+                                tracing::error!("{err:?}");
+                                continue;
+                            }
+                        };
+                    if let Err(err) = tcp_send(&mut stream, &serialized).await {
+                        tracing::warn!("failed to send reply to coordinator: {err}");
+                        continue;
+                    };
+                }
+                _ => {}
+            }
```

Comment:
So, the thing is that once we reply to the daemon through the mpsc channel. The daemon stop immediately resulting in the response to the coordinator not being sent in time and sometimes generating errors. 

See: https://github.com/dora-rs/dora/actions/runs/7534111448/job/20507893298

So, if we respond to the coordinator first. We make sure that the coordinator get a response.

Reply:
Makes sense, thanks! It still feels a bit hacky to handle the same event twice and send the reply before it is even received by the daemon main thread. Maybe it would be better to let the main thread wait until the reply is sent out before exiting? I can prepare a PR for that.

Reply:
I created a PR against this branch in https://github.com/dora-rs/dora/pull/413. Let me know what you think about this approach!

Reply:
Ok Thanks !